[hr_recruitment] Communication with candidates

miqmago opened the pull request:
Description of the issue/feature this PR addresses:
HR Recruitment Application from candidate

Current behavior before PR:
Every time a candidate sends an email to the recruitment process, it creates a new application, even if the application already exists

Desired behavior after PR is merged:
With this modification, when a candidate writes an email and his application already exists, it appends the note to all existing applications for this candidate.



--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

mart-e commented:
Hello,

Thank you for you patch but it seems you have not [signed the CLA](https://www.odoo.com/sign-cla).
Could you add it otherwise we won't be able to integrate your patch?
Regards

mart-e commented on addons/hr_recruitment/models/hr_recruitment.py:
```diff
@@ -377,6 +377,9 @@ def message_new(self, msg, custom_values=None):
             defaults['priority'] = msg.get('priority')
         if custom_values:
             defaults.update(custom_values)
+        existing = self.search([('email_from', '=', msg.get('from'))]).ids
```
the search returns a recordset, no need to use `.ids` here and `browse` later

miqmago commented:
Hi @mart-e, @appfeel has signed the cla, I can move this pull request to their account.
I've also seen some cases that email is in the form `John Doe <john.doe@example.com>` so it would be better to modify the search.

Also per your comment, would this be a better approach?

```python
            existing = self.search([
                '|',
                ('email_from', 'ilike', '<' + email_from + '>'),
                ('email_from', '=', email_from)])
            if existing:
                return existing.message_post(subtype='mail.mt_note', **msg).ids
```

mart-e commented:
> appfeel has signed the cla, I can move this pull request to their account.

The important part is the email address that has made the commit. Here you are committing with the address  `miqmago@users.noreply.github.com` while, I guess, the CLA was signed with a corprorate email. You must use the same email for the CLA signature and the commit.

>  would this be a better approach?

I let @tivisse review the content of the PR to see if accepted. I was just making a styling comment ;-)

miqmago commented:
I've moved the pull request to the user that have signed CLA: https://github.com/odoo/odoo/pull/32884

Also changed @mart-e suggestions to remove `.ids` and `browse`.
